Automattic/jetpack
# Jetpack: properly display plans with no expiration date

## Proposed changes:
* My Plan: Adjust styling for plans with no expiration date (Golden Token, Pressable Premium)

### Other information:


## Jetpack product discussion
p8oabR-1aA-p2#comment-7299
p1683321216497689-slack-C051H2TQR54
8267-gh-Automattic/jpop-issues

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect a Jetpack site on a non-Golden Token plan
2. Go to "Jetpack Dashboard -> My Plan" and confirm that the "My Plan" card looks good.
3. Use the "Jetpack Golden Tokens" tool to get your site a Golden Token. Confirm that the card looks in accordance with the design (see the Slack conversation).
